packages/vega-parser/src/parsers/axis.js

```diff
@@ -36,15 +36,22 @@ export default function(spec, scope) {
   };
   dataRef = ref(scope.add(Collect({}, [datum])));
 
+  const ariaHidden = _('ariaHidden');
+
   // encoding properties for axis group item
   axisEncode = extendEncode({
     update: {
-      offset:       encoder(_('offset') || 0),
+      offset:       encoder(_('offset', 0)),
       position:     encoder(value(spec.position, 0)),
       titlePadding: encoder(_('titlePadding')),
       minExtent:    encoder(_('minExtent')),
       maxExtent:    encoder(_('maxExtent')),
-      range:        {signal: `abs(span(range("${spec.scale}")))`}
+      range:        {signal: `abs(span(range("${spec.scale}")))`},
+      ariaHidden:   encoder(ariaHidden),
+      ariaLabel:    encoder(ariaHidden == true ? undefined : _('ariaLabel', axisAriaLabel(_, scope))),
```

Comment:
- Should either check to be truthy or use triple (not double) equals comparison.
- The ternaries can be outside the encoder calls: `ariaHidden ? undefined : encoder(_('ariaLabel', axisAriaLabel(_, scope)))`. Perhaps a cleaner approach would be to consolidate conditional logic for ariaHidden. One could ideally use object rest destructuring, but that might introduce compatibility issues with certain browsers.

Reply:
handled below together with legend